header.go

```diff
@@ -43,6 +43,7 @@ type ResponseHeader struct {
 	contentEncoding []byte
 	server          []byte
 	mulHeader       [][]byte
+	keys            [][]byte
```

Comment:
You can re-use `mulHeader` for this. But I do think we need to improve the warning for these and the `PeekAll` functions to make it clear that any further calls to `Peek*` functions will also change the return values.

Reply:
I have modified the code and added the relevant code comments